### PR TITLE
Implementing Android Jetpack Preferences

### LINKED ISSUE #3909: [Android] Implementing Android Jetpack Preferences
### Proposed title of article
[Android] Implementing Android Jetpack Preferences

### Introduction paragraph (2-3 paragraphs):
In many cases, we require a specific sort of configuration in our program. Furthermore, Android applications frequently require settings that allow the user to change the preferences in their app, which is also the ideal method to give a better user experience. To maintain or handle user preferences, Android provides a sophisticated library called Android Preference Library (JetPack). In this article, we will look at how to use the Android Preference Library (JetPack) in our project by developing a simple setting screen. 

### Key takeaways: 
By the end of this tutorial, the learner will be able to understand:
-  Different features of the Android Preference Library
- How to set up the Android Preference Library
- How to create a menu on the action bar
- How to create a Preference Screen and setting the preference in activity
- How to read the values of the settings and set a listener for settings
- 
### References: 
N/A

### PR DESCRIPTION
**Pull request by mercie-nyakio**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

This closes #3909

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:26]: Have this in paragraph form.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:37]: Highlight the steps probably i.e Step one: Adding a menu to the MainActivity

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:80]: This is xml

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:121]: This is also xml

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @mercie-nyakio   👋 .

First of all, welcome to Section - EngEd. 🎉
We are happy to have you here.

My name is Briana, I'll be reviewing your article for publishing. I will make this review as fast and as swift as possible, with your help of course 😄 .

Good job on the article. It is a very interesting read. 

Just a few changes I need you to make as we continue with the preliminary revisions:
- Please structure your article according to this [sample](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md). I need you to clearly highlight and explain your steps. Where you can add more detail please do so.
- Please upload your author info. You can find the instructions [here](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/UPLOAD_INSTRUCTIONS.md).
- Refer to the other comments I have made.

In case of any question or help needed please do not hesitate to ping me here or on Slack.
Also, notify me when you are done both here and on Slack.

Regards.

**Comment by mercie-nyakio** [comment]
hello @BrianaNzivu  I have done the requested changes.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:2]: I think this introduction can be better and a little bit more detailed.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:34]: Have this as a paragraph

**Comment by mercie-nyakio** [comment]
hello @BrianaNzivu, am done with the requested change.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (BrianaNzivu) [inline content/articles/implementing-android-jetpack-preferences/index.md:290]: You can link a Github repo here with the code for this project.

**Comment by BrianaNzivu** [comment]
Proceeding with tech review 👍

**Comment by BrianaNzivu** [comment]
Tech review done 👍 

Waiting for the minor addition I have requested for a final review to be done.

**Comment by mercie-nyakio** [comment]
hello @BrianaNzivu, I have added link to the Github repo. Please check

**Review by BrianaNzivu — APPROVED** [review]
Hey @mercie-nyakio 👋 .

Very good job on the article.  

- Please check out the resources page for tools which can help increase the quality of your writing. 
- Please go through ALL the corrections I have made so that you can not repeat the same mistakes.

That's it! :smile: 

A final review will be done now by a content moderator.
Otherwise, congratulations on publishing your article🍾.

In case of any question or clarification (currently or in the future), do not hesitate to reach out via [Slack](https://app.slack.com/client/T5X14SH7G/D01A6MT673R) or my personal email (briananzivu@gmail.com). 
I am **ALWAYS** here to help.

It has been a pleasure reading and reviewing this article.

Looking forward to seeing you contribute more to EngEd. 🚀

Kind Regards.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:17]: I don't really understand this sentence.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:19]: A whole paragraph is a single sentence. This cannot be. Break it down.

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:23]: setup rather than `setip`

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:52]: `the` rather than `he`

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:81]: In this step, (comma)

Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:81]: new `A`ndroid activity ...

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:83]: Is it the activity or folder/dir?

File: content/articles/implementing-android-jetpack-preferences/index.md
Comment (WanjaMIKE) [inline content/articles/implementing-android-jetpack-preferences/index.md:86]: Can the above be normal paragraphs

**Comment by WanjaMIKE** [comment]
Seems like there is still some work to do @mercie-nyakio

**Comment by mercie-nyakio** [comment]
Hello @WanjaMIKE, I have done the requested edits please check. Thank you
cc: @BrianaNzivu

**Comment by BrianaNzivu** [comment]
Hey @WanjaMIKE 👋 
Kindly check if the article is good to go.

Regards.

cc @mercie-nyakio

**Comment by BrianaNzivu** [comment]
> Hello @WanjaMIKE, I have done the requested edits please check. Thank you cc: @BrianaNzivu

Pinging the moderator on Slack as well 👍

**Comment by mercie-nyakio** [comment]
Hello @WanjaMIKE, Please do a final review on the article.
cc: @ahmadmardeni1, @hectorkambow

**Comment by WanjaMIKE** [comment]
Merged #6819 into master

**Pull request merged**
